ethereum/event/BREvent.c

```diff
@@ -168,48 +142,45 @@ eventHandlerAlarmCallback (BREventHandler handler,
     eventHandlerSignalEventOOB (handler, (BREvent*) &event);
 }
 
-
 typedef void* (*ThreadRoutine) (void*);
 
 static void *
 eventHandlerThread (BREventHandler handler) {
-
 #if defined (__ANDROID__)
     pthread_setname_np(handler->thread, handler->name);
```

Comment:
I was thinking on the https://breadwallet.atlassian.net/browse/CORE-564 issue and it occurs to me that that same race *could* be a problem here (never actually observed).

If we change to `pthread_setname_np(pthread_self(), handler->name);`, I think we'd be safe here 😕? Threads suck.

Reply:
That is a change needed throughout.  I changed it... but it must be assigned in the function?

Reply:
Its assigned in the thread calling `pthread_create` but this handler thread might hit that point before then (depending on cores/scheduling).